packages/amplify-util-uibuilder/src/commands/utils/detachedFormCleanup.ts

```diff
@@ -0,0 +1,46 @@
+import { StudioForm, StudioSchema } from '@aws-amplify/codegen-ui';
+import { Form } from 'aws-sdk/clients/amplifyuibuilder';
+
+/**
+ * Does the form's reference DataStore type exist in the list of models
+ */
+export const isFormDetachedFromModel = (schema: StudioForm | Form, models: Set<string>): boolean => {
+  if (schema.dataType.dataSourceType === 'DataStore' && !models.has(schema.dataType.dataTypeName)) {
+    return true;
+  }
+  return false;
+};
+
+/**
+ * A form schema will be customized if fields, style, sectionalElements objects aren't empty
+ * @param schema The schema to test
+ * @returns boolean
+ */
+export const isFormSchemaCustomized = (schema: StudioForm | Form): boolean => {
+  const { fields, style, sectionalElements } = schema;
+  // Fields and sectionalElements just need an empty check
+  if (!isEmpty({ ...fields, ...sectionalElements })) {
+    return true;
+  }
+
+  // Check style
+  if (!isEmpty(style)) {
+    // Can be empty but each style can be undefined
+    for (const styleConfig of Object.values(style)) {
+      if (styleConfig !== undefined) return true;
+    }
+  }
+
+  return false;
```

Comment:
nit: I prefer using [Arrays.prototype.some](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) here, but not a blocker

```suggestion
  if (isEmpty(style)) {
    return true; // Can be empty
  }
  // If not empty, at least one style must be defined
  return Object.values(style).some(styleConfig => styleConfig !== undefined);
```